source/tools.txt

```diff
@@ -1,168 +1,263 @@
+
+- `MongoLog <http://github.com/puentesarrin/mongodb-log/>`__ is a Python logging
+  handler that stores logs in MongoDB by using a capped collection.
+
+- `rod.recipe.mongodb <http://pypi.python.org/pypi/rod.recipe.mongodb/>`__ is a
+  `Buildout <https://www.buildout.org/en/latest/contents.html>`__ recipe for downloading
+  and installing MongoDB.
+
+Interoperability Tools
+----------------------
+
+This section lists tools that support interopability with other tools.
+
+.. _pymongo-gevent:
+
+gevent
+~~~~~~
+
+{+driver-short+} uses thread and socket functions from the Python standard library.
+By using `gevent <http://www.gevent.org/>`__, {+driver-short+}
+can do asynchronous I/O with non-blocking sockets and schedule operations
+on `greenlets <https://pypi.org/project/greenlet/>`__ instead of threads.
+
+To use gevent with {+driver-short+}, call gevent's
+``monkey.patch_all()`` method before loading any other modules, as shown in the following
+example:
+
+.. code-block:: python
+
+   >>> # You must call patch_all() *before* importing any other modules
+   >>> from gevent import monkey
+   >>> _ = monkey.patch_all()
+   >>> from pymongo import MongoClient
+   >>> client = MongoClient()
+
+.. important:: Close MongoClient to Avoid Blocking
+
+   If you call ``monkey.patch_all()`` when your application launches, ``MongoClient``
+   will use greenlets instead of threads to monitor the health of the server.
+   When shutting down, if your application calls the ``~gevent.hub.Hub.join`` method
+   without first terminating these greenlets,
+   the call to the ``~gevent.hub.Hub.join`` method blocks indefinitely.
+   
+   To avoid this, close or dereference any active ``MongoClient`` before exiting your
+   application. In some application frameworks, one solution is a signal handler
+   to end background greenlets when your application receives ``SIGHUP``:
+
+   .. code-block:: python
+
+      import signal
+
+      def graceful_reload(signum, traceback):
+           """Explicitly close some global MongoClient object."""
+           client.close()
+
+      signal.signal(signal.SIGHUP, graceful_reload)
+
+   This issue affects applications using uWSGI versions earlier than 1.9.16
+   or newer uWSGI versions with the ``-gevent-wait-for-hub`` option.
+   For more information, see
+   `the uWSGI changelog <https://uwsgi-docs.readthedocs.io/en/latest/Changelog-1.9.16.html#important-change-in-the-gevent-plugin-shutdown-reload-procedure>`__.
+
+.. _pymongo-mod_wsgi:
+
+mod-wsgi
+~~~~~~~~
+
+The `mod_wsgi <https://github.com/GrahamDumpleton/mod_wsgi/>`__ package provides an
+Apache module that implements a WSGI-compliant interface for hosting Python-based
+web applications on top of the Apache web server.
+
+To run your {+driver-short+} application under mod_wsgi, follow these guidelines:
+
+- Use the ``WSGIDaemonProcess`` directive to run mod_wsgi in daemon mode. If your
+  mod_wsgi`` configuration includes only the ``WSGIScriptAlias`` directive,
```

Comment:
I don't think this needs to be monospace, but you're missing a set of ticks if it should be.
```suggestion
  mod_wsgi configuration includes only the ``WSGIScriptAlias`` directive,
```